GitHubA11yFixes.user.js

```diff
@@ -163,6 +163,27 @@ const DYNAMIC_TWEAKS = [
 	// Remove headings from folder and file lists.
 	{selector: 'table[aria-labelledby=folders-and-files] :is(h2, h3)',
 		tweak: makePresentational},
+	// Make file viewer filenames headings, and the first item in the file viewer.
+	{selector: '.file-header .file-info .Truncate:has(.Link--primary)',
+		tweak: el => {
+			makeHeading(el, 2)
+			let headerRow = el.parentElement
+			let children = Array.from(headerRow.children)
+			// Filename is the last child of .file-info, make it the first
+			children.unshift(children.pop())
```

Comment:
I tried using `Array.prototype.indexOf` to find the index of the filename in the parent, then removing it by splicing and `unshift`ing it to the start, but it seriously slowed things down for me. This assumes the filename is the last child of `.file-info`, which it always seems to be for me.